docs/en/guides/pipeline-integrations/administration/config-staging-and-rollout.md

```diff
@@ -6,3 +6,489 @@ lang: en
+- Running command setup_project...
+
+----------------------------------------------------------------------
+Command: Setup project
+----------------------------------------------------------------------
+
+Welcome to the {% include product %} Pipeline Toolkit Project Setup!
+
+Connecting to {% include product %}...
+Connecting to the App Store...
+
+------------------------------------------------------------------
+Which configuration would you like to associate with this project?
+
+You can use the configuration from an existing project as a template for this
+new project. All settings, apps and folder configuration settings will be
+copied over to your new project. The following configurations were found:
+
+   My very first project: '/mnt/software/shotgun/first_project/config'
+
+If you want to use any of the configs listed about for your new project, just
+type in its path when prompted below.
+
+You can use the Default Configuration for your new project.  The default
+configuration is a good sample config, demonstrating a typical basic setup of
+the {% include product %} Pipeline Toolkit using the latest apps and engines. This will be
+used by default if you just hit enter below.
+
+If you have a configuration stored somewhere on disk, you can enter the path
+to this config and it will be used for the new project.
+
+You can also enter an url pointing to a git repository. Toolkit will then
+clone this repository and base the config on its content.
+
+[tk-config-default]: /mnt/software/shotgun/first_project/config
+```
+## A studio configuration in git source control
+
+Limitations with the first approach include the fact that the projects are not connected to each other. If you have 10 projects and you all need to update them because a critical bug fix has been released, you would have to manually go through each project and run the  `tank updates`  command.
+
+One way to resolve this is to create a master configuration and store it in git source control. Whenever you create a new project, simply type in the path to this git repository in the setup project dialog and toolkit will clone it for you. Now all the projects are connected to the same "studio master" config. If you have made some good changes to a project configuration, you can commit them and push them to the studio master. Other projects can then easily pull these down. You also retain a history of all your changes via git.
+
+    What is Git?
+
+![](images/config-staging-and-rollout/git_config.png)
+
+The basic idea is that you set up a git repository which holds the git configuration. Whenever you run  `tank setup_project`, you specify the git url to this repository (for example  `username@someserver.com:/studio_config.git`) and the setup process will clone the repository so that the new project basically becomes a repository connected to the main studio repository. Once they are connected you can push and pull changes, and work in branches for finer granularity.
```

Comment:
```suggestion
The basic idea is that you set up a git repository which holds the git configuration. Whenever you run `tank setup_project`, you specify the git url to this repository (for example  `username@someserver.com:/studio_config.git`) and the setup process will clone the repository so that the new project becomes a repository connected to the main studio repository. Once they are connected you can push and pull changes, and work in branches for finer granularity.
```